sdv/_utils.py

```diff
@@ -200,3 +201,80 @@ def _format_invalid_values_string(invalid_values, num_values):
             return f'{invalid_values[:num_values] + extra_missing_values}'
 
     return f'{invalid_values}'
+
+
+def _find_root_tables(relationships):
+    parent_tables = {rel['parent_table_name'] for rel in relationships}
+    child_tables = {rel['child_table_name'] for rel in relationships}
+    return parent_tables - child_tables
+
+
+def _get_relationship_for_child(relationships, child_table):
+    return [rel for rel in relationships if rel['child_table_name'] == child_table]
+
+
+def _get_relationship_idx_for_parent(relationships, parent_table):
+    return [
+        idx for idx, rel in enumerate(relationships) if rel['parent_table_name'] == parent_table
```

Comment:
Same thing, do we need the index here or can we just pass the relationship?

Reply:
Yes, done in [f110a87](https://github.com/sdv-dev/SDV/pull/1800/commits/f110a87f22a179943bf4dba242824de8d6ad4cec)